.github/workflows/_cleanup.yml

```diff
@@ -0,0 +1,62 @@
+name: Cleanup
+
+on:
+  workflow_call:
+    inputs:
+      cluster-name:
+        required: true
+        description: AKS Cluster Name With Randomly Generated Suffix
+        type: string
+  workflow_dispatch:
```

Comment:
Instead of a workflow_dispatch (which allows manually calling it). I would add a cron job and see if you can make the cluster cleanup something as simple as "Delete all clusters" when they're ready. This will allow for a nightly cleanup of the resource group and can ensure everything is deleted in case a pipeline failure causes something to avoid cleanup. See the aci examples cleanup file for ideas on how this might be done: https://github.com/microsoft/confidential-aci-examples/blob/main/.github/workflows/cleanup.yml

schedule:
    - cron: "0 0 * * *"

Reply:
done